iroh-net/src/dns.rs

```diff
@@ -97,10 +100,69 @@ pub trait ResolverExt {
         host: N,
         timeout: Duration,
     ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
+
+    /// Looks up node info by DNS name.
+    fn lookup_by_domain(&self, name: &str) -> impl Future<Output = Result<NodeAddr>>;
+
+    /// Looks up node info by [`NodeId`] and origin domain name.
+    fn lookup_by_id(
+        &self,
+        node_id: &NodeId,
+        origin: &str,
+    ) -> impl Future<Output = Result<NodeAddr>>;
+
+    /// Perform an ipv4 lookup with a timeout in a staggered fashion.
+    ///
+    /// The first call is done immediately, with added calls according to the staggering strategy.
+    fn staggered_lookup_ipv4<N: IntoName + Clone>(
```

Comment:
Since I'm in a naming nitpicking mood, maybe `lookup_ipv4_staggered` is the right thing here as it leaves the `lookup_` as prefix?
(I'm less sure about `lookup_ipv4_staggered` vs `lookup_staggered_ipv4`)

Reply:
I like the reasoning for `lookup_`, I think it makes it easier to find if you just have the lsp suggesting code. Done in 373e172